### ISSUE TITLE
warning: enumeration values 'AllocatedPointer' and 'AllocAlign' not handled in switch

### ISSUE THREAD

**Issue by est31** [rust-lang/rust#99955]
On recent master, compiling gives me this warning in stdout:

```
warning: llvm-wrapper/RustWrapper.cpp:159:11: warning: enumeration values 'AllocatedPointer' and 'AllocAlign' not handled in switch [-Wswitch]
warning:   switch (Kind) {
warning:           ^
warning: 1 warning generated.
```

Now, they *are* handled in the switch case, but are gated by `LLVM_VERSION_GE`. I think this is a fall out of #99574. The issue is probably fixed by adding similar gating to LLVMWrapper.h .

@rustbot label A-llvm T-compiler
